lib/PerMessageDeflate.js

```diff
@@ -76,10 +76,18 @@ PerMessageDeflate.prototype.accept = function(paramsList) {
 
 PerMessageDeflate.prototype.cleanup = function() {
   if (this._inflate) {
+    if (this._inflate.writeInProgress) {
+      this._inflate.pendingClose = true;
+      return;
```

Comment:
I don't know this codebase, but... I think this means that if there is an ```_inflate.writeInProgress``` when the connection is closed, the ```_deflate``` will never be closed. Instead of returning, this should be rewritten as:
```
if (this._inflate.writeInProgress) {
      this._inflate.pendingClose = true;
} else {
      if (this._inflate.close) this._inflate.close();
     this._inflate = null;
}
```